src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/commands/order/AddOrderIngredientCommand.java

```diff
@@ -0,0 +1,123 @@
+package ay2122s1_cs2103t_w16_2.btbb.logic.commands.order;
+
+import static ay2122s1_cs2103t_w16_2.btbb.commons.util.CollectionUtil.requireAllNonNull;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.commands.order.AddOrderCommand.MESSAGE_DUPLICATE_ORDER;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.commands.util.CommandUtil.makeOrderWithEditedIngredients;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_INGREDIENT_NAME;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_INGREDIENT_QUANTITY;
+import static ay2122s1_cs2103t_w16_2.btbb.logic.parser.util.CliSyntax.PREFIX_INGREDIENT_UNIT;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ay2122s1_cs2103t_w16_2.btbb.commons.core.Messages;
+import ay2122s1_cs2103t_w16_2.btbb.commons.core.index.Index;
+import ay2122s1_cs2103t_w16_2.btbb.exception.CommandException;
+import ay2122s1_cs2103t_w16_2.btbb.exception.NotFoundException;
+import ay2122s1_cs2103t_w16_2.btbb.logic.commands.Command;
+import ay2122s1_cs2103t_w16_2.btbb.logic.commands.CommandResult;
+import ay2122s1_cs2103t_w16_2.btbb.logic.descriptors.IngredientDescriptor;
+import ay2122s1_cs2103t_w16_2.btbb.model.Model;
+import ay2122s1_cs2103t_w16_2.btbb.model.ingredient.Ingredient;
+import ay2122s1_cs2103t_w16_2.btbb.model.order.Order;
+import ay2122s1_cs2103t_w16_2.btbb.model.order.RecipeIngredientList;
+import ay2122s1_cs2103t_w16_2.btbb.ui.UiTab;
+
+/**
+ * Adds an ingredient to an existing order in btbb.
+ */
+public class AddOrderIngredientCommand extends Command {
+    public static final String COMMAND_WORD = "add-oi";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an ingredient to the order identified "
+            + "by the index number used in the displayed order list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_INGREDIENT_NAME + "NAME "
+            + PREFIX_INGREDIENT_QUANTITY + "QUANTITY "
+            + PREFIX_INGREDIENT_UNIT + "UNIT\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_INGREDIENT_NAME + "Rice "
+            + PREFIX_INGREDIENT_QUANTITY + "400 "
+            + PREFIX_INGREDIENT_UNIT + "g";
+
+    public static final String MESSAGE_ADD_ORDER_INGREDIENT_SUCCESS = "Added Order Ingredient: %1$s\n"
+            + "Edited Order: %2$s";
```

Comment:
Should we inform the user of the change in the inventory as well? To make it obvious that ingredient quantity in inventory has changed or not.